File: clients/java/src/main/java/com/svix/diom/apis/AuthToken.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.ApiException;
import com.svix.diom.HttpClient;
import com.svix.diom.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import lombok.Getter;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.diom.models.AuthTokenCreateIn;
import com.svix.diom.models.AuthTokenCreateOut;
import com.svix.diom.models.AuthTokenDeleteIn;
import com.svix.diom.models.AuthTokenDeleteOut;
import com.svix.diom.models.AuthTokenExpireIn;
import com.svix.diom.models.AuthTokenExpireOut;
import com.svix.diom.models.AuthTokenListIn;
import com.svix.diom.models.AuthTokenUpdateIn;
import com.svix.diom.models.AuthTokenUpdateOut;
import com.svix.diom.models.AuthTokenVerifyIn;
import com.svix.diom.models.AuthTokenVerifyOut;
import com.svix.diom.models.ListResponseAuthTokenOut;

public class AuthToken {
    private final HttpClient client;

    public AuthToken(HttpClient client) {
        this.client = client;
    }

    /** Create Auth Token */
    public AuthTokenCreateOut create(
        final AuthTokenCreateIn authTokenCreateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/create");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenCreateIn,
            AuthTokenCreateOut.class
        );
    }

    /** Expire Auth Token */
    public AuthTokenExpireOut expire(
        final AuthTokenExpireIn authTokenExpireIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/expire");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenExpireIn,
            AuthTokenExpireOut.class
        );
    }

    /** Delete Auth Token */
    public AuthTokenDeleteOut delete(
        final AuthTokenDeleteIn authTokenDeleteIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/delete");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenDeleteIn,
            AuthTokenDeleteOut.class
        );
    }

    /** Verify Auth Token */
    public AuthTokenVerifyOut verify(
        final AuthTokenVerifyIn authTokenVerifyIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/verify");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenVerifyIn,
            AuthTokenVerifyOut.class
        );
    }

    /** List Auth Tokens */
    public ListResponseAuthTokenOut list(
        final AuthTokenListIn authTokenListIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/list");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenListIn,
            ListResponseAuthTokenOut.class
        );
    }

    /** Update Auth Token */
    public AuthTokenUpdateOut update(
        final AuthTokenUpdateIn authTokenUpdateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/auth-token/update");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            authTokenUpdateIn,
            AuthTokenUpdateOut.class
        );
    }
}

File: clients/javascript/src/apis/authToken.ts
// this file is @generated

import {
    type AuthTokenCreateIn,
    AuthTokenCreateInSerializer,
} from '../models/authTokenCreateIn';
import {
    type AuthTokenCreateOut,
    AuthTokenCreateOutSerializer,
} from '../models/authTokenCreateOut';
import {
    type AuthTokenDeleteIn,
    AuthTokenDeleteInSerializer,
} from '../models/authTokenDeleteIn';
import {
    type AuthTokenDeleteOut,
    AuthTokenDeleteOutSerializer,
} from '../models/authTokenDeleteOut';
import {
    type AuthTokenExpireIn,
    AuthTokenExpireInSerializer,
} from '../models/authTokenExpireIn';
import {
    type AuthTokenExpireOut,
    AuthTokenExpireOutSerializer,
} from '../models/authTokenExpireOut';
import {
    type AuthTokenListIn,
    AuthTokenListInSerializer,
} from '../models/authTokenListIn';
import {
    type AuthTokenUpdateIn,
    AuthTokenUpdateInSerializer,
} from '../models/authTokenUpdateIn';
import {
    type AuthTokenUpdateOut,
    AuthTokenUpdateOutSerializer,
} from '../models/authTokenUpdateOut';
import {
    type AuthTokenVerifyIn,
    AuthTokenVerifyInSerializer,
} from '../models/authTokenVerifyIn';
import {
    type AuthTokenVerifyOut,
    AuthTokenVerifyOutSerializer,
} from '../models/authTokenVerifyOut';
import {
    type ListResponseAuthTokenOut,
    ListResponseAuthTokenOutSerializer,
} from '../models/listResponseAuthTokenOut';
import { AuthTokenNamespace } from './authTokenNamespace';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class AuthToken {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    public get namespace() {
        return new AuthTokenNamespace(this.requestCtx);
    }

    /** Create Auth Token */
    public create(
        authTokenCreateIn: AuthTokenCreateIn,
    ): Promise<AuthTokenCreateOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/auth-token/create");

        request.setBody(
            AuthTokenCreateInSerializer._toJsonObject(authTokenCreateIn)
        );
        
        return request.send(
            this.requestCtx,
            AuthTokenCreateOutSerializer._fromJsonObject,
        );
    }/** Expire Auth Token */
    public expire(
        authTokenExpireIn: AuthTokenExpireIn,
    ): Promise<AuthTokenExpireOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/auth-token/expire");

        request.setBody(
            AuthTokenExpireInSerializer._toJsonObject(authTokenExpireIn)
        );
        
        return request.send(
            this.requestCtx,
            AuthTokenExpireOutSerializer._fromJsonObject,
        );
    }/** Delete Auth Token */
    public delete(
        authTokenDeleteIn: AuthTokenDeleteIn,
    ): Promise<AuthTokenDeleteOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/auth-token/delete");

        request.setBody(
            AuthTokenDeleteInSerializer._toJsonObject(authTokenDeleteIn)
        );
        
        return request.send(
            this.requestCtx,
            AuthTokenDeleteOutSerializer._fromJsonObject,
        );
    }/** Verify Auth Token */
    public verify(
        authTokenVerifyIn: AuthTokenVerifyIn,
    ): Promise<AuthTokenVerifyOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/auth-token/verify");

        request.setBody(
            AuthTokenVerifyInSerializer._toJsonObject(authTokenVerifyIn)
        );
        
        return request.send(
            this.requestCtx,
            AuthTokenVerifyOutSerializer._fromJsonObject,
        );
    }/** List Auth Tokens */
    public list(
        authTokenListIn: AuthTokenListIn,
    ): Promise<ListResponseAuthTokenOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/auth-token/list");

        request.setBody(
            AuthTokenListInSerializer._toJsonObject(authTokenListIn)
        );
        
        return request.send(
            this.requestCtx,
            ListResponseAuthTokenOutSerializer._fromJsonObject,
        );
    }/** Update Auth Token */
    public update(
        authTokenUpdateIn: AuthTokenUpdateIn,
    ): Promise<AuthTokenUpdateOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/auth-token/update");

        request.setBody(
            AuthTokenUpdateInSerializer._toJsonObject(authTokenUpdateIn)
        );
        
        return request.send(
            this.requestCtx,
            AuthTokenUpdateOutSerializer._fromJsonObject,
        );
    }
}

